test/runtests.jl

```diff
@@ -134,5 +134,33 @@ for (h, h_name) ∈ ((NormL1(λ), "l1"),)
   end
 end
 
+R2N_R2DH(args...; kwargs...) = R2N(args...; subsolver = R2DH, kwargs...)
+for (mod, mod_name) ∈ ((SpectralGradientModel, "spg"), (DiagonalPSBModel, "psb"), (LSR1Model, "lsr1"), (LBFGSModel, "lbfgs"))
+  for (h, h_name) ∈ ((NormL0(λ), "l0"), (NormL1(λ), "l1"))
+    for solver_sym ∈ (:R2DH, :R2N, :R2N_R2DH)
+      solver_sym == (:R2N,:R2N_R2DH)  && mod_name == ("spg", "psb") && continue
```

Comment:
@MohamedLaghdafHABIBOULLAH did you mean
```suggestion
      solver_sym ∈ (:R2N,:R2N_R2DH)  && mod_name ∈ ("spg", "psb") && continue
```
  ?  Is it expected for R2N to fail/perform worse on diagonal hessian approximations ?

Reply:
This one handle exception for R2DH: which expects diagonal update form, while R2N expects LBFGS, LSR1, or exact operators.